tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Services/ProfileServiceTests.cs

```diff
@@ -0,0 +1,377 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Moq;
+using NHSD.BuyingCatalogue.Identity.Api.Constants;
+using NHSD.BuyingCatalogue.Identity.Api.Models;
+using NHSD.BuyingCatalogue.Identity.Api.Repositories;
+using NHSD.BuyingCatalogue.Identity.Api.UnitTests.Builders;
+using NUnit.Framework;
+using static IdentityModel.JwtClaimTypes;
+using static NHSD.BuyingCatalogue.Identity.Api.Constants.ApplicationClaimTypes;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.UnitTests.Services
+{
+    [TestFixture]
+    public sealed class ProfileServiceTests
+    {
+        [Test]
+        public async Task GetProfileDataAsync_GivenAnApplicationUserExists_ReturnsExpectedClaimList()
+        {
+            var expectedApplicationUser = ApplicationUserBuilder
+                .Create()
+                .WithEmail("TestUser@Email.com")
+                .WithFirstName("Bob")
+                .WithLastName("Smith")
+                .WithOrganisationFunction("Authority")
+                .Build();
+
+            Mock<IUserRepository> applicationUserRepositoryMock = new Mock<IUserRepository>();
+            applicationUserRepositoryMock.Setup(x => x.FindByIdAsync(It.IsAny<string>()))
+                .ReturnsAsync(expectedApplicationUser);
+
+            var sut = ProfileServiceBuilder
+                .Create()
+                .WithUserRepository(applicationUserRepositoryMock.Object)
+                .Build();
+
+            var profileDataRequestContext = ProfileDataRequestContextBuilder
+                .Create()
+                .WithSubjectId(expectedApplicationUser.Id)
+                .Build();
+
+            await sut.GetProfileDataAsync(profileDataRequestContext);
+
+            var expected = new List<KeyValuePair<string, string>>
```

Comment:
using KeyValuePair here is a bit nasty, it adds a lot of noise. We could do something like:

var expected = new List<(string, string)>{
  (Subject, expectedApplicationUser.Id),
  (PreferredUserName, expectedApplicationUser.UserName)
};

var actual = blah.Select(item => (item.Type, item.Value))
actual.Should().BeEquivilentTo(expected);

Reply:
Yeah good shout on this one.